eng/common/scripts/Verify-RestApiSpecLocation.ps1

```diff
@@ -0,0 +1,275 @@
+      LogError "ServiceDir:$ServiceDirectory, PackageName:$PackageName. Invalid repo in the file url: $fileUrl. Repo should be 'Azure/azure-rest-api-specs'."
+      exit 1
+    }
+    # check the commit hash belongs to main branch
+    Verify-CommitFromMainBranch $commit
+  }
+  else{
+    LogError "ServiceDir:$ServiceDirectory, PackageName:$PackageName. Invalid file url: $fileUrl"
+    exit 1
+  }
+}
+
+# This function is used to verify the 'repo' and 'commit' settings in tsp-location.yaml point to the main branch of Azure/azure-rest-api-specs repository
+function Verify-TspLocation([System.Object]$tspLocationObj) {
+  $repo = $tspLocationObj["repo"]
+  $commit = $tspLocationObj["commit"]
+  if($repo -ne "Azure/azure-rest-api-specs") {
+    LogError "Invalid repo setting in the tsp-location.yaml: $repo. Repo should be 'Azure/azure-rest-api-specs'. ServiceDir:$ServiceDirectory, PackageName:$PackageName"
+    exit 1
+  }
+
+  # check the commit hash belongs to main branch
+  Verify-CommitFromMainBranch $commit
+}
+
+# This function is used to verify the specific 'commit' belongs to the main branch of Azure/azure-rest-api-specs repository
+function Verify-CommitFromMainBranch([string]$commit) {
+  if($commit -notmatch "^[0-9a-f]{40}$" -and $commit -ne "main") {
+    LogError "Invalid commit hash or branch name: $commit. Branch name should be 'main' or the commit should be a 40-character SHA-1 hash. ServiceDir:$ServiceDirectory, PackageName:$PackageName"
+    exit 1
+  }
+  if($commit -eq "main") {
+    Write-Host "ServiceDir:$ServiceDirectory, PackageName:$PackageName. Branch is $commit branch of Azure/azure-rest-api-specs repository."
+    return
+  }
+  try {
+    $searchResult = Search-GitHubCommit -AuthToken $GitHubPat -CommitHash $commit -RepoOwner "Azure" -RepoName "azure-rest-api-specs"
+    if ($searchResult.total_count -lt 1) {
+      LogError "Commit $commit doesn't exist in 'main' branch of Azure/azure-rest-api-specs repository. ServiceDir:$ServiceDirectory, PackageName:$PackageName"
+      exit 1
+    }
+    else{
+      Write-Host "ServiceDir:$ServiceDirectory, PackageName:$PackageName. Commit $commit exists in 'main' branch of Azure/azure-rest-api-specs repository."
+    }
+  }
+  catch {
+    LogError "ServiceDir:$ServiceDirectory, PackageName:$PackageName. Failed to search commit $commit with exception:`n$_ "
+    exit 1
+  }
+}
+
+function Verify-YamlContent([string]$markdownContent) {
+  $splitString = '``` yaml|```yaml|```'
+  $yamlContent = $markdownContent -split $splitString
+  foreach($yamlSection in $yamlContent) {
+    if ($yamlSection) {
+      try {
+        # remove the lines like: $(tag) == 'package-preview-2023-09'
+        $yamlSection = $yamlSection -replace '^\s*\$\(.+\)\s*==.+', ''
+        $yamlobj = ConvertFrom-Yaml -Yaml $yamlSection
+        if($yamlobj) {
+          $batchValue = $yamlobj["batch"]
+          $requireValue = $yamlobj["require"]
+          $inputFileValue = $yamlobj["input-file"]
```

Comment:
Are batchValue, requireValue, inputFileValue exclusive in a readme file? If they can exist concurrently, there will be problem.

Reply:
I haven't found a real case with having two of these configurations. Currently, I follow below sequence for the validation. 
Do you have any idea of how does the generation tool handle this? 
+@weidongxu-microsoft, any comment?

Reply:
I am OK with current order.
In my understanding, we are not achieving full support for the markdown for autorest. (one may need to use the parse code from autorest to achieve full support, e.g. https://github.com/Azure/autorest/blob/main/packages/libs/configuration/src/literate-yaml/literate-yaml.test.ts)

So, as long as it works reasonably fine, I am good with it.

Reply:
Thanks for sharing the reference of autorest code logic. I learned that autorest handles the case of `json` content in the markdown file. I would like to ignore that part because I haven't found the usage of `json` in `autorest.md` and `swagger/README.md`.